src/main/java/Duke.java

```diff
@@ -1,10 +1,145 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
+        String taskType;
+        String taskName;
+        String by;
+        String at;
+        Scanner splitInputScanner;
+        String[] userInputSplitted = {""};
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        // To do textUITesting, comment line 19
+        System.out.println(logo);
+        printDividingLine();
+
+        // Print welcome message
+        printWelcomeMessage();
+
+        // Scan for input
+        Scanner userInputScanner = new Scanner(System.in);
+        String userInput = userInputScanner.nextLine();
+
+        // Separate taskType and taskName(may contain
+        // dates/time) if applicable
+        if(isOneWord(userInput)) {
+            taskType = userInput;
+            taskName = userInput;   // dummy value
+        }
+        else{
+            splitInputScanner = new Scanner(userInput);
+            taskType = splitInputScanner.next();
+            taskName = splitInputScanner.nextLine();
+        }
+
+        // Loop for user input until "bye" is inputted
+        while(!taskType.equals("bye")) {
+            switch (taskType) {
+            case "todo":
+                Task t = new Todo(taskName);
+                printDividingLine();
+                t.addTask();
+                printDividingLine();
+                break;
+            case "deadline":
+                by = extractTime(taskName);
+                taskName  = extractTaskName(taskName);
+                Task d = new Deadline(taskName, by);
+                printDividingLine();
+                d.addTask();
+                printDividingLine();
+                break;
+            case "event":
+                at = extractTime(taskName);
+                taskName = extractTaskName(taskName);
+                Task e = new Event(taskName, at);
+                //e.setTime(time);
+                printDividingLine();
+                e.addTask();
+                printDividingLine();
+                break;
+            case "list":
+                printDividingLine();
+                Task.listTasks();
+                printDividingLine();
+                break;
+            case "done":
+                printDividingLine();
+                Task.markAsDone(getTaskIndex(userInput));
+                printDividingLine();
+                break;
+            default:
+                printCommandErrorMessage();
+                break;
```

Comment:
Would it be better if you tweak the code slightly so that you can extract methods to avoid duplicate codes?